### PR TITLE
Optimise mcontext init by avoiding excessive zeroing.

### PR DESCRIPTION
**Pull request by abhinav92003**
Instead of using partial struct initialisation, simply initialise the required fields.

### CODE REVIEW

File: api/samples/cbr.c [hunk 1]
```diff
@@ -232,10 +232,9 @@ insert(hash_table_t table, app_pc addr, cbr_state_t state)
 static void
 at_taken(app_pc src, app_pc targ)
 {
-    dr_mcontext_t mcontext = {
-        sizeof(mcontext),
-        DR_MC_ALL,
-    };
+    dr_mcontext_t mcontext;
```
Comment (derekbruening) [inline api/samples/cbr.c:235]: optional: Do you think a reader might wonder why we're doing it this way?  If it were a small struct using {} is simpler.  We've found that the time taken to zero the large struct shows up as noticeable overhead for optimized clients.  Maybe a simple comment to make that clear?

Comment (derekbruening) [inline api/samples/cbr.c:235]: commit message style: No "IX: " in the title (just on the branch name) -- so the title would just start with "Opt..."

**Review by derekbruening — APPROVED** [review]
Thank you for the cleanup.  We discovered the zeroing of the big struct was a perf hit a while back on Dr. Memory and somehow I thought we had already changed all our samples to set good examples.

File: api/samples/cbr.c (hunk 1 above)
Comment (abhinav92003) [reply api/samples/cbr.c:240]: Done. Added the comment in just `api/samples/cbr.c`, not in the (many) tests.

Comment (abhinav92003) [reply api/samples/cbr.c:240]: Thanks. Done.

**Pull request merged**